onnxruntime/core/optimizer/rule_based_graph_transformer.cc

```diff
@@ -25,7 +25,7 @@ Status RuleBasedGraphTransformer::ApplyRulesOnNode(Graph& graph, Node& node,
   return Status::OK();
 }
 
-Status TopDownRuleBasedTransformer::ApplyImpl(Graph& graph, bool& modified, int graph_level) const {
+Status RuleBasedGraphTransformer::ApplyImpl(Graph& graph, bool& modified, int graph_level) const {
```

Comment:
>RuleBasedGraphTransformer [](start = 7, length = 25)

Maybe we could make this more generic and the ctor could take a value saying whether we recurse first or last.